File: udp/cliente.py
import socket
import json

def send_message(host: str, port: int, message:str):
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    server_socket.sendto(message, (host, port))

if __name__=='__main__':
    HOST = '10.20.22.186' #10.20.22.186 - end do professor
    PORT = 9000

name = input ('Enter your name: ')

while True:
    message = input ('Type your message: ')

    body_message = f"{name} != {message}".encode('utf-8')
    
    # print(body_message)

    send_message(HOST, PORT, body_message)

File: udpp/cliente.js
const dgram = require('dgram');
const readline = require('readline');

const client = dgram.createSocket('udp4');
const PORT = 9000;
const HOST = 'localhost';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter your name: ', (name) => {
    function promptMessage() {
        rl.question('Type your message: ', (msg) => {
            const bodyMessage = `${name}!=${msg}`;
            client.send(bodyMessage, PORT, HOST, (err) => {
                if (err) console.error(err);
                promptMessage();
            });
        });
    }
    promptMessage();
});
